### PR TITLE
Added Flex Counters support for tunnel counters

### PR DESCRIPTION
**Pull request by dgsudharsan**
Added Flex counter support for  tunnel counters and rates implementation

### CODE REVIEW

File: syncd/FlexCounter.cpp [hunk 1]
```diff
@@ -114,6 +114,14 @@ FlexCounter::MACsecSAAttrIds::MACsecSAAttrIds(
     // empty intentionally
 }
 
+FlexCounter::TunnelCounterIds::TunnelCounterIds(
+        _In_ sai_object_id_t tunnel,
+        _In_ const std::vector<sai_tunnel_stat_t> &tunnelIds):
+    tunnelId(tunnel), tunnelCounterIds(tunnelIds)
```
Comment (kcudnik) [inline syncd/FlexCounter.cpp:120]: put each variable in separate line, also name members with m_ prefix

File: syncd/FlexCounter.cpp [hunk 2]
```diff
@@ -114,6 +114,14 @@ FlexCounter::MACsecSAAttrIds::MACsecSAAttrIds(
...
+{
+    SWSS_LOG_ENTER();
+}
```
Comment (kcudnik) [inline syncd/FlexCounter.cpp:123]: add // empty
to notify that this is empty intentionally

**Review by kcudnik — CHANGES_REQUESTED** [review]

File: syncd/FlexCounter.cpp [hunk 3]
```diff
@@ -114,6 +114,14 @@ FlexCounter::MACsecSAAttrIds::MACsecSAAttrIds(
     // empty intentionally
 }
 
+FlexCounter::TunnelCounterIds::TunnelCounterIds(
+        _In_ sai_object_id_t tunnel,
```
Comment (kcudnik) [inline syncd/FlexCounter.cpp:118]: tunnel vid or rid?

File: syncd/FlexCounter.cpp (hunk 1 above)
Comment (dgsudharsan) [reply syncd/FlexCounter.cpp]: Done

File: syncd/FlexCounter.cpp (hunk 2 above)
Comment (dgsudharsan) [reply syncd/FlexCounter.cpp:125]: Done

File: syncd/FlexCounter.cpp (hunk 3 above)
Comment (dgsudharsan) [reply syncd/FlexCounter.cpp]: tunnelRid. I modified variable name to reflect it.

**Comment by kcudnik** [comment]
seems like build its failing because it requires your changes from swss-common

**Comment by dgsudharsan** [comment]
@kcudnik The pipeline is passing now. Could you please help merging the PR?

**Review by kcudnik — APPROVED** [review]

**Pull request merged**